rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1599,34 +1710,29 @@ public NodeSet getExactlyNodes(int nb, SelectionScript selectionScript) {
      * @return RMInitialState containing nodes and nodeSources of the RMCore.
      */
     public RMInitialState getRMInitialState() {
-        ArrayList<RMNodeEvent> nodesList = new ArrayList<>(this.allNodes.values().size());
 
-        Map<String, RMNodeEvent> nodeEvents = new HashMap<>();
+        Map<String, RMNodeEvent> nodeEvents = new HashMap<>(this.allNodes.size());
```

Comment:
This getRMInitialState could be on streams like this:
```
 final Map<String, RMNodeEvent> nodeEvents = this.allNodes
                .values()
                .stream()
                .map(rmNode -> rmNode.createNodeEvent())
                .collect(Collectors.toMap(event -> event.getNodeUrl(), event -> event));
```

Reply:
Alright, I also updated it for the node source events.